src/meta/processors/admin/CreateSnapshotProcessor.cpp

```diff
@@ -0,0 +1,63 @@
+/* Copyright (c) 2019 vesoft inc. All rights reserved.
+ *
+ * This source code is licensed under Apache 2.0 License,
+ * attached with Common Clause Condition 1.0, found in the LICENSES directory.
+ */
+
+
+#include "meta/processors/admin/CreateSnapshotProcessor.h"
+#include "meta/processors/admin/SnapShot.h"
+
+DEFINE_int32(checkpoint_delay_secs, 30 , "wal synchronization delay");
+
+namespace nebula {
+namespace meta {
+using SignType = storage::cpp2::EngineSignType;
+
+void CreateSnapshotProcessor::process(const cpp2::CreateSnapshotReq& req) {
+    UNUSED(req);
+    auto snapshot = currentSnapshot();
+    folly::SharedMutex::WriteHolder wHolder(LockUtils::snapshotLock());
+
+    auto signRet = Snapshot::instance(kvstore_)->sendBlockSign(SignType::BLOCK_ON);
+    if (signRet != cpp2::ErrorCode::SUCCEEDED) {
+        resp_.set_code(cpp2::ErrorCode::E_SNAPSHOT_FAILURE);
+        onFinished();
+        return;
+    }
+
+    kvstore_->setBlocking(true);
```

Comment:
We should block all writes on every storage host before make snapshot.

Reply:
Yes, the all storages blocked via line 22.